contracts/evoting/evoting.go

```diff
@@ -821,6 +834,108 @@ func (e evotingCommand) manageAdminForm(snap store.Snapshot, step execution.Step
 	return nil
 }
 
+// TODO modify
+// manageVotersForm implements commands. It performs the ADD or REMOVE VOTERS command
+func (e evotingCommand) manageVotersForm(snap store.Snapshot, step execution.Step) error {
+	msg, err := e.getTransaction(step.Current)
+	if err != nil {
+		return xerrors.Errorf(errGetTransaction, err)
+	}
+
+	var form types.Form
+	var formID []byte
+
+	txAddVoter, okAddVoter := msg.(types.AddVoter)
+	txRemoveVoter, okRemoveVoter := msg.(types.RemoveVoter)
+
+	if okAddVoter {
+		form, formID, err = e.getForm(txAddVoter.FormID, snap)
+		if err != nil {
+			return xerrors.Errorf(errGetForm, err)
+		}
+
+		err = form.AddVoter(txAddVoter.UserID)
+		if err != nil {
+			return xerrors.Errorf("couldn't add voter: %v", err)
+		}
```

Comment:
the only difference between these two possibe transactions and the actions to add/remove owners is the error message no? could you do something like (pseudocode, didn't test it)

```
switch(msg) {
    case types.AddVoter {
          operation = form.AddVoter
          err_msg = "couldn't add voter: %v"
    }
    ...
}
```
?

Reply:
and then the actual execution:

```
form, formID, err = e.getForm(tx.FormID, snap)
...
```
since all these transactions share the same data model?